### ISSUE TITLE
`-AignoreDeadCode`: detect branches a single checker's own dataflow proves unreachable, not just CFG-structural dead code

### ISSUE THREAD

**Issue by wmdietl** [eisop/checker-framework#2131]
## Summary

`-AignoreDeadCode` (added in #2128, closing #627) currently detects two kinds of dead code:
- CFG-structural unreachability (e.g. a `catch` block whose `try` body cannot throw), via the restored `isUnreachable`/`reachableNodes` mechanism.
- Syntactically compile-time-constant `if` conditions (`if (true)`/`if (false)`/a `static final boolean` field), via `BaseTypeVisitor.visitIf`.

Neither mechanism catches a branch that's dead only because *the running checker's own type-system refinement* proves the branch condition can never hold. Two examples, both currently still reported as errors under `-AignoreDeadCode`:

```java
// Nullness checker: objOut's refined type is exactly "null", so the loop body never runs.
Object objOut = null;
while (objOut != null) {
    objInner.toString();  // still reported, should be suppressed
}
```

```java
// Value checker: i's refined range at loop entry is exactly {0}, so the loop body never runs.
for (int i = 0; i < 0; i++) {
    obj.toString();  // still reported, should be suppressed
}
```

These look superficially different (nullness vs. numeric reasoning) but are the same underlying gap: `reachableNodes` is populated once from the CFG's structure, before dataflow's fixed point runs, and is never informed by the abstract values (`Store`/`Value`) the analysis itself computes. It never asks "is the incoming store for this block even satisfiable?" — only "is this block reachable in the unannotated CFG?" Running only the Value checker wouldn't catch the first example, and running only the Nullness checker wouldn't catch the second, which shows this isn't checker-specific: whichever single checker is running should be able to use its own refined stores to prune branches its own analysis proves dead.

## Proposed scope

**First iteration:** for a single running checker, treat a CFG block whose only incoming stores are refined-impossible (per that checker's own `Store`/`Value` semantics) as unreachable, and feed that into the existing `reachableNodes`/`isUnreachable` machinery alongside the current CFG-structural facts.

**Deliberately deferred:** combining refinements across multiple simultaneously-running checkers (e.g., a Value-checker-derived numeric fact pruning a branch only the Nullness checker's diagnostics would otherwise flag). That's a distinct, harder problem — it requires cross-checker cooperation infrastructure that doesn't exist today — and should only be tackled if/when checkers are actually composed for this purpose.

## Test cases

Both of the above are captured (currently documented as a limitation) as `test2`/`test3` in eisop#633's `checker/tests/nullness-ignoredeadcode/DeadBranchNPE.java`; that PR's approach (patching individual error-reporting call sites) was superseded by #2128's choke-point approach, but the test scenarios themselves are still open once this issue is addressed.
